File: abc213/d.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> ans;
vector<vector<int>> path;

void dfs(int now, int pre) {
  ans.push_back(now);
  for(auto to: path[now]) {
    if (to != pre) {
      dfs(to, now);
      ans.push_back(now);
    }
  }
}


int main() {
  int n;
  cin >> n;
  path.resize(n+1);
  for(int i=0; i<n-1; i++){
    int a, b;
    cin >> a >> b;
    path[a].push_back(b);
    path[b].push_back(a);
  }
  for(int i=0; i<n+1; i++){
    sort(path[i].begin(), path[i].end());
  }
  dfs(1, -1);
  for(auto elem: ans) {
    cout << elem << " ";
  }
  cout << endl;

  
  return 0;
}

File: abc213/d.py
import sys
sys.setrecursionlimit(10**6)
ans = []
def dfs(now, pre):
    ans.append(now)
    for to in path[now]:
        if to != pre:
            dfs(to, now)
            ans.append(now)

n = int(input())
path = [[] for _ in range(n+1)]
for _ in range(n-1):
    a, b = map(int, input().split())
    path[a].append(b)
    path[b].append(a)
for i in range(n+1): path[i].sort()
dfs(1, -1)
print(*ans)
